k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -1830,7 +1830,7 @@ void processPendingAppend(PendingAppend command, Account account) throws Messagi
                     if (remoteDate != null) {
                         remoteMessage.setFlag(Flag.DELETED, true);
                         if (Expunge.EXPUNGE_IMMEDIATELY == account.getExpungePolicy()) {
-                            remoteFolder.expunge();
+                            remoteFolder.expunge(localFolder.getAllMessageUids(true));
```

Comment:
With my suggestions applied this should only pass the UID of the current message to `expungeUids()`.